src/ensembl/src/content/app/entity-viewer/gene-view/components/protein-domain-image/ProteinDomainImage.tsx

```diff
@@ -59,20 +68,33 @@ export const getDomainsByResourceGroups = (
 };
 
 const ProteinDomainImage = (props: ProteinDomainImageProps) => {
-  const { start: transcriptStart } = getFeatureCoordinates(props.transcript);
+  const [data, setData] = useState<Product | null>(null);
 
-  if (!props.transcript.product.protein_domains_resources) {
+  useEffect(() => {
+    fetchTranscript(props.transcriptId).then((result) => {
+      if (result?.product) {
+        setData(result.product);
+      }
+    });
```

Comment:
Probably also set guard against updating the state of an unmounted component:

```
  useEffect(() => {
    let canUpdate = true;
    fetchTranscript(props.transcriptId).then((result) => {
      if (result?.product && canUpdate) {
        setData(result.product);
      }
    });
    return () => canUpdate = false;
}, [props.transcriptId])
```
```

Reply:
I added `AbortController` to cancel the API calls

Reply:
I've introduced a `useApiService` hook in another PR (https://github.com/Ensembl/ensembl-client/pull/288/files#diff-4f114f907b791696cc9840ba7a5539b3), which should probably be concerned with all that low-level stuff.

But OK.